### PR TITLE
Opsgenie additional commands

### PR DESCRIPTION
**Pull request by gshriki**
add opsgenie-get-schedules and opsgenie-get-schedule-timelines commands to get further information from opsgenie

### CODE REVIEW

**Comment by dorsha** [comment]
@gshriki lets go over this together tomorrow

File: Integrations/integration-OpsGenie.yml [hunk 1]
```diff
@@ -38,8 +38,9 @@ script:
         users: 'users'
     };
 
-    var sendRequest = function(url) {
-        var requestUrl = baseURL + url;
+    var sendRequest = function(url, param) {
```
Comment (dorsha) [inline Integrations/integration-OpsGenie.yml]: you added the `param` parameter as second argument.
I checked all `sendRequest` calls, and it seems that you never pass the second argument... am I missing something? If not in use, please remove.

File: Integrations/integration-OpsGenie.yml [hunk 2]
```diff
@@ -63,24 +64,30 @@ script:
...
-    var getOnCall = function(schedule) {
-        var url = serviceURLs.schedules + '/' + encodeURIComponent(schedule) + '/on-calls?scheduleIdentifierType=name';
+    var getOnCall = function(schedule, date) {
+        var urlParams = {scheduleIdentifierType:'name'};
+        if (date) {
+            urlParams.date = new Date(date).toISOString();
+        }
+        var url = serviceURLs.schedules + '/' + encodeURIComponent(schedule) + '/on-calls' + encodeToURLQuery(urlParams);
+
+        console.log(url);
```
Comment (dorsha) [inline Integrations/integration-OpsGenie.yml]: remove this console.log

File: Integrations/integration-OpsGenie.yml [hunk 3]
```diff
@@ -93,27 +100,65 @@ script:
...
+        } else {
+            md = 'No schedules\n';
+        }
+        return {Type: entryTypes.note, Contents: res.data, EntryContext: ec, ContentsFormat: formats.json, HumanReadable: md};
+    };
+
+    var getScheduleTimeline = function(schedule) {
+        var url = serviceURLs.schedules + '/' + schedule + '/timeline?identifierType=name';
+        var res = sendRequest(url);
+        var rotations = res.data.finalTimeline.rotations;
```
Comment (dorsha) [inline Integrations/integration-OpsGenie.yml]: you are checking below for `(res && res.data)` but you access `res.data` without checking.. you might have issue here. 
Just move this line inside the if statement

**Comment by dorsha** [comment]
@gshriki let's fix and merge. should be quick.

File: Integrations/integration-OpsGenie.yml (hunk 1 above)
Comment (gshriki) [reply Integrations/integration-OpsGenie.yml]: you are right. had it FFU, but no need for now.

File: Integrations/integration-OpsGenie.yml (hunk 2 above)
Comment (gshriki) [reply Integrations/integration-OpsGenie.yml]: removed

File: Integrations/integration-OpsGenie.yml (hunk 3 above)
Comment (gshriki) [reply Integrations/integration-OpsGenie.yml]: yep... fixed.

**Pull request merged**